src/linechart/DataSeries.jsx

```diff
@@ -2,139 +2,100 @@
-    var voronoiSeriesRef = props.structure.reference(['voronoiSeries', props.seriesName]);
-
-
-    var circles = null;
-
-    if (props.displayDataPoints) {
-      // Map over data to generate SVG circles at data points
-      // if datum is a date object, treat it a bit differently
-      circles = props.data.map(function(point, idx) {
-        var cx, cy;
-        if (this._isDate(point, xAccessor)) {
-          cx = props.xScale(xAccessor(point).getTime());
-        } else {
-          cx = props.xScale(xAccessor(point));
-        }
-        if (this._isDate(point, yAccessor)) {
-          cy = props.yScale(yAccessor(point).getTime());
+        if (this._isDate(props.data[0].values[0], xAccessor)) {
+          interpolatePath.x(function(d) {
+            return props.xScale(props.xAccessor(d).getTime());
+          });
         } else {
-          cy = props.yScale(yAccessor(point));
+          interpolatePath.x(function(d) {
+            return props.xScale(props.xAccessor(d));
+          });
         }
 
-        var id = props.seriesName + '-' + idx;
-
-        // Create an immstruct reference for the circle id
-        // and set it to 'inactive'
-        props.structure.cursor('voronoi').set(id, 'inactive');
-
-        // Having set the Voronoi circle id cursor to 'inactive'
-        // We now pass on the Voronoi circle id reference to the
-        // circle component, where it will be observed and dereferenced
-        var voronoiRef = props.structure.reference(['voronoi', id]);
-
-        return (
-          <Circle
-            voronoiRef={voronoiRef}
-            voronoiSeriesRef={voronoiSeriesRef}
-            structure={props.structure}
-            cx={cx}
-            cy={cy}
-            r={props.circleRadius}
-            fill={props.fill}
-            key={idx}
-            id={props.seriesName + '-' + idx}
+    var lines = props.data.map((series, idx) => {
+      return <Line 
+        path={interpolatePath(series.values)}
+        stroke={props.colors(props.colorAccessor(series, idx))}
+        seriesName={series.name}
+        key={idx}
+      />
+    });
+
+    var voronoi = d3.geom.voronoi()
+      .x(function(d){ return xScale(d.coord.x); })
+      .y(function(d){ return yScale(d.coord.y); })
+      .clipExtent([[0, 0], [ props.width , props.height]]);
+
+    var cx, cy, circleFill;
+    var regions = voronoi(props.value).map(function(vnode, idx) {
+      var point = vnode.point.coord;
+      if (Object.prototype.toString.call(xAccessor(point)) === '[object Date]') {
+        cx = props.xScale(xAccessor(point).getTime());
+      } else {
+        cx = props.xScale(xAccessor(point));
+      }
+      if (Object.prototype.toString.call(yAccessor(point)) === '[object Date]') {
+        cy = props.yScale(yAccessor(point).getTime());
+      } else {
+        cy = props.yScale(yAccessor(point));
+      }
+      circleFill = props.colors(props.colorAccessor(vnode, vnode.point.seriesIndex));
+      
```

Comment:
Imho, since `var circleFill` is only used a single place, just get rid of it and call `props.colors(props.colorAccessor(vnode, vnode.point.seriesIndex))` directly when you build the `VoronoiCircleContainer`

```javascript
<VoronoiCircleContainer 
    key={idx} 
    circleFill={props.colors(props.colorAccessor(vnode, vnode.point.seriesIndex))}
    circleRadius={props.circleRadius}
    cx={cx} 
    cy={cy}
     vnode={vnode}
/>
```